src/Aspire.Dashboard/Telemetry/DashboardTelemetrySender.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Threading.Channels;
+
+namespace Aspire.Dashboard.Telemetry;
+
+public class DashboardTelemetrySender : IDashboardTelemetrySender
+{
+    private readonly HttpClient _httpClient;
+
+    private readonly Channel<(List<Guid>, Func<HttpClient, Func<Guid, object>, Task<ICollection<object>>>)> _channel = Channel.CreateUnbounded<(List<Guid>, Func<HttpClient, Func<Guid, object>, Task<ICollection<object>>>)>();
+    private readonly ConcurrentDictionary<Guid, object> _responsePropertyMap = [];
+
+    public DashboardTelemetrySender(HttpClient client, ILogger<IDashboardTelemetryService> logger)
+    {
+        _httpClient = client;
+        _ = Task.Run(async () =>
+        {
+            while (await _channel.Reader.WaitToReadAsync().ConfigureAwait(false))
+            {
+                while (_channel.Reader.TryRead(out var operation))
+                {
+                    var (guids, requestFunc) = operation;
+                    try
+                    {
+                        var result = await requestFunc(client, GetResponseProperty).ConfigureAwait(false);
+                        foreach (var (guid, value) in guids.Zip(result))
+                        {
+                            _responsePropertyMap[guid] = value;
+                        }
+                    }
+                    catch (Exception ex) when (ex is HttpRequestException or JsonException or ArgumentException)
+                    {
+                        logger.LogWarning("Failed to make telemetry request: {ExceptionMessage}", ex.Message);
+                    }
+
+                    continue;
+
+                    object GetResponseProperty(Guid guid)
```

Comment:
guid is a type, not the name of a variable. Is this a property id? Call it `propertyId`.